test/RibbonThetaSTETHVault.ts

```diff
@@ -0,0 +1,2907 @@
+
+    describe("#redeem", () => {
+      time.revertToSnapshotAfterEach();
+
+      it("reverts when 0 passed", async function () {
+        await assetContract
+          .connect(userSigner)
+          .approve(vault.address, depositAmount);
+        await vault.depositETH({ value: depositAmount });
+        await rollToNextOption();
+        await expect(vault.redeem(0)).to.be.revertedWith("!shares");
+      });
+
+      it("overflows when shares >uint104", async function () {
+        const redeemAmount = BigNumber.from(
+          "340282366920938463463374607431768211455"
+        );
+        await assetContract
+          .connect(userSigner)
+          .approve(vault.address, depositAmount);
+        await vault.depositETH({ value: depositAmount });
+        await rollToNextOption();
+        await expect(vault.redeem(redeemAmount)).to.be.revertedWith(">U104");
+      });
+
+      it("reverts when redeeming more than available", async function () {
+        await assetContract
+          .connect(userSigner)
+          .approve(vault.address, depositAmount);
+        await vault.depositETH({ value: depositAmount });
+
+        await rollToNextOption();
+
+        await expect(vault.redeem(depositAmount.add(1))).to.be.revertedWith(
+          "Exceeds available"
+        );
+      });
+
+      it("decreases unredeemed shares", async function () {
+        await assetContract
+          .connect(userSigner)
+          .approve(vault.address, depositAmount);
+        await vault.depositETH({ value: depositAmount });
+
+        await rollToNextOption();
+
+        const redeemAmount = BigNumber.from(1);
+        const tx1 = await vault.redeem(redeemAmount);
+
+        await expect(tx1)
+          .to.emit(vault, "Redeem")
+          .withArgs(user, redeemAmount, 1);
+
+        const {
+          round: round1,
+          amount: amount1,
+          unredeemedShares: unredeemedShares1,
+        } = await vault.depositReceipts(user);
+
+        assert.equal(round1, 1);
+        assert.bnEqual(amount1, BigNumber.from(0));
+        assert.bnEqual(unredeemedShares1, depositAmount.sub(redeemAmount));
+
+        const tx2 = await vault.redeem(depositAmount.sub(redeemAmount));
+
+        await expect(tx2)
+          .to.emit(vault, "Redeem")
+          .withArgs(user, depositAmount.sub(redeemAmount), 1);
+
+        const {
+          round: round2,
+          amount: amount2,
+          unredeemedShares: unredeemedShares2,
+        } = await vault.depositReceipts(user);
+
+        assert.equal(round2, 1);
+        assert.bnEqual(amount2, BigNumber.from(0));
+        assert.bnEqual(unredeemedShares2, BigNumber.from(0));
+      });
+    });
+
+    describe("#withdrawInstantly", () => {
+      let minETHOut: BigNumberish;
+
+      time.revertToSnapshotAfterEach(async function () {
+        const crv = await getContractAt("ICRV", STETH_ETH_CRV_POOL);
+        minETHOut = (await crv.get_dy(1, 0, depositAmount))
+          .mul(crvSlippage.add(100))
```

Comment:
for some reason transaction reverts on crv.get_dy(1, 0, depositAmount) do you have any idea why? I checked on the actual contract https://etherscan.io/address/0xDC24316b9AE028F1497c275EB9192a3Ea0f67022#readContract and everything works...

Reply:
Nothing stands out to me why it reverts..

Reply:
Does the pool exist when the block is forked

Reply:
13056027 is 3 days ago...